frontend_lib/src/helper.js

```diff
@@ -390,7 +388,73 @@ export const serialize = (objectToSerialize, propertyMap) => {
   )
 }
 
-export const getCurrentContentVersionNumber = (appFeatureMode, content, timeline) => {
+export const getCurrentContentVersionNumber = async (appFeatureMode, content, timeline) => {
   if (appFeatureMode === APP_FEATURE_MODE.REVISION) return content.number
   return timeline.filter(t => t.timelineType === 'revision' && t.hasBeenRead).length
 }
+
+export const MINIMUM_CHARACTERS_USERNAME = 3
+export const MAXIMUM_CHARACTERS_USERNAME = 255
+export const ALLOWED_CHARACTERS_USERNAME = 'azAZ09-_'
+
+// Check that the given username is valid.
+// Return an object:
+// {isUsernameValid: false, usernameInvalidMsg: 'Username invalid'}
+// The message is translated using the given props.t.
+export const checkUsernameValidity = async (apiUrl, username, props) => {
+  if (username.length < MINIMUM_CHARACTERS_USERNAME) {
+    return {
+      isUsernameValid: false,
+      usernameInvalidMsg: props.t('Username must be at least {{minimumCharactersUsername}} characters long', { minimumCharactersUsername: MINIMUM_CHARACTERS_USERNAME })
+    }
+  }
+
+  if (username.length > MAXIMUM_CHARACTERS_USERNAME) {
+    return {
+      isUsernameValid: false,
+      usernameInvalidMsg: props.t('Username must be at maximum {{maximumCharactersUsername}} characters long', { maximumCharactersUsername: MAXIMUM_CHARACTERS_USERNAME })
+    }
+  }
+
+  if (hasSpaces(username)) {
+    return {
+      isUsernameValid: false,
+      usernameInvalidMsg: props.t("Username can't contain any whitespace")
+    }
+  }
+
+  // INFO - GB - 2020-06-08 The allowed characters are azAZ09-_
+  if (!(/^[A-Za-z0-9_-]*$/.test(username))) {
+    return {
+      isUsernameValid: false,
+      usernameInvalidMsg: props.t('Allowed characters: {{allowedCharactersUsername}}', { allowedCharactersUsername: ALLOWED_CHARACTERS_USERNAME })
+    }
+  }
+
+  const fetchReservedUsernames = await getReservedUsernames(apiUrl)
+  if (fetchReservedUsernames.status !== 200 || !fetchReservedUsernames.json.items) {
+    throw new Error(props.t('Error while checking reserved usernames'))
+  }
+  if (fetchReservedUsernames.json.items.indexOf(username) >= 0) {
```

Comment:
```suggestion
  if (fetchReservedUsernames.json.items.indexOf(username) >== 0) {
```

Reply:
? I do not think this operator exists (at least firefox and node.js give me a syntax error if I use it in a test).